[Low Zheng Hui] iP

zhenghuil opened the pull request:
# Reacher, helps u remember your tasks to do.
## functions
- add `add`
- edit
1. delete
2. list
for example,
```
add
edit
```
**- [x] mark done**
_- [ ] edit_
- [ ] add
🦅 
> this is a quote
[this is a link](https://github.com/nus-cs2103-AY2324S2/ip/pull/392)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

zhenghuil committed:
named reacher and greet

zhenghuil committed:
deleted Duke.java

zhenghuil committed:
added echo

zhenghuil committed:
added add and list function

zhenghuil committed:
added function to change task done status and task class

zhenghuil committed:
added new 3 task subclasses

zhenghuil committed:
handled exceptions

zhenghuil committed:
added delete

zhuuyicheng commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markDone() {
+        this.isDone = true;
+    }
+    public void markNotDone() {
+        this.isDone = false;
+    }
```
I like that your names representing methods are verbs and written in camelCase, good job following the coding standards.

zhuuyicheng commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+import java.util.SplittableRandom;
+
+public class Deadline extends Task {
+    protected String by;
```
I like that your class variables are never declared public, in accordance to the coding standards 👍

zhuuyicheng commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good job naming your boolean variable to sound like a boolean as per the coding standards.

zhuuyicheng commented on src/main/java/Reacher.java:
```diff
@@ -0,0 +1,91 @@
+                if (input.isEmpty()) {
+                    throw new ReacherException("pls type a task name.");
+                }
+                if (input.equalsIgnoreCase("bye")) { //check for end request
```
While the if/else statements work properly to identify and separate logic blocks for different inputs, maybe you could also look into using switch/case statements or even enums which can look neater since there are so many different commands?

zhuuyicheng reviewed (commented):
Overall, LGTM.

breezetall commented on src/main/java/Reacher.java:
```diff
@@ -0,0 +1,91 @@
+
+public class Reacher {
+    public static void main(String[] args) {
+            Scanner scanner = new Scanner(System.in);
+        ArrayList<Task> memory = new ArrayList<>();
```
Should these be indented the same?

breezetall replied:
Perhaps a more intuitive name such as taskList?

breezetall commented on src/main/java/Reacher.java:
```diff
@@ -0,0 +1,91 @@
+                } else if (input.equalsIgnoreCase("list")) {// check for list request
+                    System.out.println("Tasks:");
+                    int c = 1;
+                    for (Task task : memory) {
```
Perhaps a normal for loop would be better here? Then the declaration of c would be within the same scope as the for loop.

breezetall commented on src/main/java/Reacher.java:
```diff
@@ -0,0 +1,91 @@
+                        System.out.println(e.getMessage());
+                    }
+                }
+            } catch (Exception e) {
```
I like that you added a catch Exception to catch all the exceptions that might have gone past ReacherException!

breezetall reviewed (commented):
Overall, I think this code is well-written. Just some minor coding standard and naming issues.

zhenghuil committed:
add storage class

zhenghuil committed:
add storage class

zhenghuil committed:
Add functionality of storage

record of tasks is stored in arraylist

every time the program is restarted, task record is lost.

lets add Storage class to manage writing and loading tasklist from a
text file so that the tasks are recorded in the computer hardware and
will not be lost after exiting program.

zhenghuil committed:
Change date from String to LocalDate

Add logic to prevent start of event after the end

zhenghuil committed:
Merge branch 'branch-Level-8'

zhenghuil committed:
ignore storage.txt

zhenghuil committed:
Add Parser, TaskList, Ui classes

zhenghuil committed:
add function to print message by ui
change from arraylist to tasklist

zhenghuil committed:
add command class
Use Ui class for user input

zhenghuil committed:
add parser class for logic

zhenghuil committed:
Add Command class, AddCommand, ExitCommand, ListCommand, EditCommand
class

zhenghuil committed:
Merge branch 'A-MoreOOP'

zhenghuil committed:
package all classes into reacher

zhenghuil committed:
add reacher.task and reacher.command packages

zhenghuil committed:
Merge branch 'add-gradle-support'

zhenghuil committed:
set up gradle

zhenghuil committed:
override equals method for all command classes
add parserTest class

zhenghuil committed:
Add UiTest class to test readString method

zhenghuil committed:
Merge branch 'A-JUnit'

zhenghuil committed:
Add JavaDoc comments

zhenghuil committed:
Edit to comply with coding standard

zhenghuil committed:
Add FindCommand class

zhenghuil committed:
Add Find function

zhenghuil committed:
Add catch DateTimeParseException in Reacher.java

zhenghuil committed:
Merge tag 'A-JavaDoc'

zhenghuil committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/reacher/Storage.java
#	src/main/java/reacher/Ui.java
#	src/main/java/reacher/command/ExitCommand.java
#	src/main/java/reacher/command/ListCommand.java
#	src/main/java/reacher/task/Task.java

zhenghuil committed:
Merge tag 'Level-9'

zhenghuil committed:
Add support for JavaFx

zhenghuil committed:
working fxml without response

zhenghuil committed:
change parser to parse full commands
add gui

zhenghuil committed:
Merge branch 'try'

zhenghuil committed:
add intro message

zhenghuil committed:
Change getInfo 2nd parameter to use enum

2nd parameter takes an int i and method returns the ith word from input.
Difficult to use as one has to remember which position the required info
is in.

Let's create an enum class for each type of info the user can give and
match the type to the position so that method can take in the enum class
as parameter and return the relevant info.

zhenghuil committed:
Remove Ui class and moved toString method to TaskList instead.

zhenghuil committed:
Change showIntro to add ReacherDialog instead of Userdialog.

zhenghuil committed:
Command.java: Remove isExit() method as it is not used
AddCommand.java: Add line breaks for readability

zhenghuil committed:
Change javaDoc heading for all Command classes to return a String
Main.java: Add reacherDialog in showIntro() for readability

zhenghuil committed:
Refactor execute command by pulling out duplicated code in switch clause

zhenghuil committed:
Add assert in default switch case

zhenghuil committed:
Add assert in default switch as at least one case is correct

zhenghuil committed:
bug fix: remove . from Execute() javaDoc

zhenghuil committed:
Add help command

There is no info on how to use bot when user first start it.

New users would not know the specific command structure and has to
resort to trial and error which is not user-friendly.

Lets add a helpCommand that executes when users enter "help"
which displays the command structure for every command available
so that users can immediately start using it.

zhenghuil committed:
Merge pull request #2 from zhenghuil/A-CodeQuality

A code quality

zhenghuil committed:
Merge branch 'master' of https://github.com/zhenghuil/ip

* 'master' of https://github.com/zhenghuil/ip:
  bug fix: remove . from Execute() javaDoc
  Add assert in default switch as at least one case is correct
  Refactor execute command by pulling out duplicated code in switch clause
  Change javaDoc heading for all Command classes to return a String Main.java: Add reacherDialog in showIntro() for readability
  Command.java: Remove isExit() method as it is not used AddCommand.java: Add line breaks for readability
  Change showIntro to add ReacherDialog instead of Userdialog.
  Remove Ui class and moved toString method to TaskList instead.
  Change getInfo 2nd parameter to use enum
  add intro message

zhenghuil committed:
Merge branch 'BCD-Extension'

zhenghuil committed:
Merge branch 'A-Assertions'

# Conflicts:
#	src/main/java/reacher/command/AddCommand.java

zhenghuil committed:
Bug fix: remove ui class form helpCommand

zhenghuil committed:
resolve merge conflict

zhenghuil committed:
Add changes to introMessage

zhenghuil committed:
Add title to Gui window

zhenghuil committed:
Bug fix: added break in switch clause

zhenghuil committed:
Add break in switch clause

zhenghuil committed:
Change main class in Gradle
Change error message for wrong task

zhenghuil committed:
Add files via upload

zhenghuil committed:
Update README.md

zhenghuil committed:
Bug fix

zhenghuil committed:
Merge branch 'master' of https://github.com/zhenghuil/ip
